src/main/java/vg/civcraft/mc/civmodcore/world/operations/ChunkOperationManager.java

```diff
@@ -0,0 +1,133 @@
+	public static final ChunkOperationManager INSTANCE = new ChunkOperationManager();
+	private static final CivModCorePlugin PLUGIN = CivModCorePlugin.getInstance();
+	private static final Map<WorldXZ, List<ChunkOperation>> STORAGE = new HashMap<>();
+
+	@EventHandler
+	public void onChunkLoad(final ChunkLoadEvent event) {
+		final Chunk chunk = event.getChunk();
+		final World world = chunk.getWorld();
+		final WorldXZ wxz = new WorldXZ(world.getUID(), chunk.getX(), chunk.getZ());
+		final List<ChunkOperation> operations = STORAGE.get(wxz);
+		if (Iteration.isNullOrEmpty(operations)) {
+			return;
+		}
+		for (final ChunkOperation operation : operations) {
+			executeOperation(operation, chunk);
+		}
+		operations.clear();
+	}
+
+	@CommandAlias("chunkoperations")
+	@Description("Shows a zoomed out view of all pending chunk operations.")
+	@CommandPermission("cmc.debug")
+	public void onDebugCommand(final CommandSender sender) {
+		sender.sendMessage(ChatColor.YELLOW + "Chunk Operations:");
+		for (final Map.Entry<WorldXZ, List<ChunkOperation>> entry : STORAGE.entrySet()) {
+			final WorldXZ wxz = entry.getKey();
+			final List<ChunkOperation> operations = entry.getValue();
+			final int operationCount = operations == null ? 0 : operations.size();
+			final World world = WorldTracker.getLoadedWorld(wxz.getWorld());
+			// First line: " WorldName: 22 operations"
+			final TextComponent textComponent = new TextComponent(" ");
+			if (world == null) {
+				textComponent.addExtra(TextUtil.textComponent("{NOT LOADED}", ChatColor.RED));
+			}
+			else {
+				textComponent.addExtra(TextUtil.textComponent(world.getName(), ChatColor.GREEN));
+			}
+			textComponent.addExtra(": ");
+			if (operationCount == 0) {
+				textComponent.addExtra("0 operations");
+				sender.spigot().sendMessage(textComponent);
+				continue;
+			}
+			else if (operationCount == 1) {
+				textComponent.addExtra("1 operation");
+			}
+			else {
+				textComponent.addExtra(operationCount + " operations");
+			}
+			sender.spigot().sendMessage(textComponent);
+			// Other lines: " - PluginName: 1 operation"
+			final Map<Plugin, Integer> pluginCount = new HashMap<>();
+			for (final ChunkOperation operation : operations) {
+				pluginCount.computeIfPresent(operation.plugin, (o, count) -> count + 1);
+				pluginCount.putIfAbsent(operation.plugin, 0);
+			}
+			for (final Map.Entry<Plugin, Integer> count : pluginCount.entrySet()) {
+				sender.sendMessage(" - " + ChatColor.AQUA + count.getKey().getName() + ": " +
+						ChatColor.RESET + count.getValue());
+			}
+		}
+		sender.sendMessage(ChatColor.RED + "End of operations.");
+	}
+
+	/**
+	 * Stages an operation for a particular chunk, which may be executed immediately if the chunk is loaded.
+	 *
+	 * @param location The location of the chunk to operate on. MUST NOT BE A BLOCK LOCATION!
+	 * @param operation The operation to perform.
+	 */
+	public static void stageOperation(final Location location, final ChunkOperation operation) {
```

Comment:
Should return a boolean indicating whether the action was immediately executed or not, maybe even a CompleteableFuture or something like it